### PR TITLE
Wayland (as seen on "edge")

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: snap/hooks/configure [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+    if [ -e "$SNAP_DATA/config_options" ]; then
+        echo "ERROR: 'snapctl' is missing, configuration will fail."
+        echo "    This should be fixed (in core18) soon."
+        echo "    Workaround: set app to one of:"
+        ls -1 $SNAP/apps/ | sed "s/^/        /"
+        echo "    by editing /var/snap/mir-kiosk-apps/current/config_options"
+        exit 1
+    fi
+fi
+
```
Comment (Saviq) [inline snap/hooks/configure:14]: I think that's unnecessary now.

File: glue/bin/mir-kiosk-app-daemon [hunk 2]
```diff
@@ -23,27 +23,12 @@ else
...
-fi
-
 # XKB config
 export XKB_CONFIG_ROOT=${SNAP}/usr/share/X11/xkb
 
-# Qt Platform to Mir
-export QT_QPA_PLATFORM=ubuntumirclient
+# Qt Platform to Wayland
+export QT_QPA_PLATFORM=wayland
+export QT_QPA_PLATFORM_PLUGIN_PATH=$(find ${SNAP}/usr/lib/ -name qt5)/plugins/platforms/
```
Comment (Saviq) [inline glue/bin/mir-kiosk-app-daemon:31]: Is this here to determine the arch triplet? If so, you can use:

```yaml
commands:
…
    environment:
      QT_QPA_PLATFORM_PLUGIN_PATH=${SNAP}/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/plugins/platforms/
```

**Review by Saviq — CHANGES_REQUESTED** [review]
Couple of nits.

File: glue/bin/mir-kiosk-app-daemon (hunk 2 above)
Comment (AlanGriffiths) [reply glue/bin/mir-kiosk-app-daemon:31]: You can't - $SNAPCRAFT variables are no longer available at runtime.

(Yes, this and similar code can be cleaned up, but let's do that separately.)

File: snap/hooks/configure (hunk 1 above)
Comment (AlanGriffiths) [reply snap/hooks/configure:14]: Yes

File: glue/bin/mir-kiosk-app-daemon (hunk 2 above)
Comment (Saviq) [reply glue/bin/mir-kiosk-app-daemon:31]: They get evaluated at build time though. You can't use them in the wrapper, but you can - in `snapcraft.yaml`.

Comment (AlanGriffiths) [reply glue/bin/mir-kiosk-app-daemon:31]: Yes, I'm preparing a separate PR for that

File: snap/hooks/configure (hunk 1 above)
Comment (Saviq) [reply snap/hooks/configure:14]: Actually I was wrong, it's not in the stable `core18` base yet.

**Review by Saviq — APPROVED** [review]
Yes.

**Pull request merged**